src/vehicle.h

```diff
@@ -552,6 +616,12 @@ class vehicle : public JsonSerializer, public JsonDeserializer
     // Translate mount coords "p" into tile coords "q" using given pivot direction and anchor
     void coord_translate (int dir, const point &pivot, const point &p, point &q) const;
 
+    /** Get all vehicle parts (if any) at @ref pos optionally including @ref broken parts */
+    static std::vector<vehicle_part *> get_parts( const tripoint &pos, bool broken = false );
+
+    /** Get first part (if any) at @ref pos which matches the predicate @ref func */
+    static vehicle_part *get_part( const tripoint &pos, const std::function<bool(const vehicle_part *)>& func );
```

Comment:
Why does the callback work on a pointer? Does `get_part` ever pass a null pointer to the callback? Why does the actual callback (e.g. in `vehicle::turret_query`) not check for null? If the callback never gets a null pointer, it should take a reference, which implicitly promises that the argument will never be null.